pyaedt/edb.py

```diff
@@ -3356,3 +3358,147 @@ def calculate_initial_extent(self, expansion_factor):
+            edb_zones[edb_zone_path] = poly_data
+        return edb_zones
+
+    @pyaedt_function_handler()
+    def cutout_multizone_layout(self, zone_dict, common_reference_net):
+        """Create multizone project cutout.
+
+        Parameters
+        ----------
+        zone_dict : dict[str](EDB PolygonData)
+            Dictionary with EDB path as key and EDB PolygonData as value defining the zone region.
+            This dictionary is returned from the command copy_zones():
+            >>> edb = Edb(edb_file)
+            >>> zone_dict = edb.copy_zones(r"C:\Temp\test")
+
+        common_reference_net : str
+            the common reference net name. This net name must be provided to provide a valid project.
+
+        Returns
+        -------
+        dict[str][str] , list of str
+        first dictionary defined_ports with edb name as key and existing port name list as value. Those ports are the
+        ones defined before processing the multizone clipping.
+        second is the list of connected port.
+
+        """
+        terminals = {}
+        defined_ports = {}
+        for edb_path, zone_polygon in zone_dict.items():
+            edb = Edb(edbversion=self.edbversion, edbpath=edb_path)
+            edb.stackup.stackup_mode = "Laminate"
+            signal_nets = list(self.nets.signal.keys())
+            edb.cutout(use_pyaedt_cutout=True, custom_extent=zone_polygon, open_cutout_at_end=True)
+            edb.active_cell.SetName(os.path.splitext(os.path.basename(edb_path))[0])
+            defined_ports[os.path.splitext(os.path.basename(edb_path))[0]] = list(edb.excitations.keys())
+            edb_terminals_info = edb.hfss.create_vertical_circuit_port_on_clipped_traces(
+                nets=signal_nets, reference_net=common_reference_net, user_defined_extent=zone_polygon
+            )
+            if edb_terminals_info:
+                terminals[os.path.splitext(os.path.basename(edb_path))[0]] = edb_terminals_info
+            edb.save_edb()
+            edb.close_edb()
+        project_connexions = self._get_connected_ports_from_multizone_cutout(terminals)
+        return defined_ports, project_connexions
+
+    @pyaedt_function_handler()
+    def _get_connected_ports_from_multizone_cutout(self, terminal_info_dict):
+        """Return connected port list from clipped multizone layout.
+
+        Parameters
+            terminal_info_dict : dict[str][str]
+                dictionary terminals with edb name as key and created ports name on clipped signal nets.
+                Dictionary is generated by the command cutout_multizone_layout:
+                >>> edb = Edb(edb_file)
+                >>> edb_zones = edb.copy_zones(r"C:\Temp\test")
+                >>> defined_ports, terminals_info = edb.cutout_multizone_layout(edb_zones, common_reference_net)
+                >>> project_connexions = get_connected_ports(terminals_info)
+
+        Returns
+        -------
+        list[str]
+            list of connected ports.
```

Comment:
```suggestion
            List of connected ports.
```
The data type in the first line is lower case but the description always starts with a capital letter.